streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -137,45 +130,43 @@
     void to(Serde<K> keySerde, Serde<V> valSerde, String topic);
 
     /**
-     * Applies a stateful transformation to all elements in this stream.
+     * Create a new {@link KStream} instance by applying a stateful transformation to all elements in this stream.
      *
-     * @param transformerSupplier the class of valueTransformerSupplier
-     * @param stateStoreNames the names of the state store used by the processor
-     * @return the instance of KStream that contains transformed keys and values
+     * @param transformerSupplier   the instance of {@link TransformerSupplier} that generates stateful processors
```

Comment:
Why the qualification that says "that generates stateful processors"?  AFAIU nothing prevents the user from supplying a state-less transformer (supplier), even though yes, you can also provide the names of state stores when calling this method.